### PR TITLE
Better vfu_attach_ctx() UT

### PR DESCRIPTION
**Pull request by swapnili**
Signed-off-by: Swapnil Ingle <swapnil.ingle@nutanix.com>

### CODE REVIEW

File: test/unit-tests.c [hunk 1]
```diff
@@ -345,15 +345,26 @@ test_realize_ctx(void **state __attribute__((unused)))
     assert_null(vfu_ctx.pci.caps);
 }
 
+static int
+dummy_attach(vfu_ctx_t *vfu_ctx)
```
Comment (tmakatos) [inline test/unit-tests.c:349]: It's simpler to define this function nested in `test_attach_ctx`, this way you'll have visibility of the actual `vfu_ctx` argument.

File: test/unit-tests.c [hunk 2]
```diff
@@ -345,15 +345,26 @@ test_realize_ctx(void **state __attribute__((unused)))
...
 static void
 test_attach_ctx(void **state __attribute__((unused)))
 {
+    struct transport_ops transport_ops = {
+        .attach = &dummy_attach,
+    };
     vfu_ctx_t vfu_ctx = {
-        .trans = &sock_transport_ops,
+        .trans = &transport_ops,
         .fd = 111
```
Comment (tmakatos) [inline test/unit-tests.c:364]: no need to initialize `.fd` now

**Review by tmakatos — APPROVED** [review]
Approved, minor nits.

File: test/unit-tests.c (hunk 1 above)
Comment (swapnili) [reply test/unit-tests.c]: Done

File: test/unit-tests.c (hunk 2 above)
Comment (swapnili) [reply test/unit-tests.c]: Done

File: test/unit-tests.c [hunk 3]
```diff
@@ -348,12 +348,21 @@ test_realize_ctx(void **state __attribute__((unused)))
 static void
 test_attach_ctx(void **state __attribute__((unused)))
 {
+    int dummy_attach(vfu_ctx_t *vfu_ctx)
+    {
+        assert(vfu_ctx != NULL);
```
Comment (tmakatos) [inline test/unit-tests.c:353]: What I meant was to define `vfu_ctx` before `dummy_attach` , and then check in `dummy_attach` that the passed argument contains the correct value, .e.g

```
...
vfu_ctx_t vfu_ctx;

int dummy_attach(vfu_ctx_t *_vfu_ctx) {
    assert_ptr_equal(_vfu_ctx == &vfu_ctx);
    return 0xdeadbeef;
}
...
```
Anyway that's minor stuff.

**Review by tmakatos — APPROVED** [review]

**Pull request merged**